### PR TITLE
Shipping Simulator + useLazyQuery

### PR DESCRIPTION
**Pull request by lucis**
[PR Marin](https://github.com/vtex-sites/marinbrasil.store/pull/94)

### CODE REVIEW

File: packages/gatsby-theme-vtex/src/sdk/graphql/useLazyQuery.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { request, RequestOptions } from '@vtex/gatsby-plugin-graphql'
+import useSWR, { ConfigInterface } from 'swr'
+
+type QueryOptions = ConfigInterface & RequestOptions
+
+// TODO: Change here
+const getKey = (options: QueryOptions) =>
+  `${options.sha256Hash}::${JSON.stringify(options.variables)}`
+
+export const useLazyQuery = <
```
Comment (tlgimenes) [inline packages/gatsby-theme-vtex/src/sdk/graphql/useLazyQuery.ts:10]: can't we have the same feature set with the raw useSWR ?

File: packages/gatsby-theme-vtex/src/components/ShippingSimulator/index.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  skuId,
+  seller,
+  country,
+  initialPostalCode,
+  variant,
+}) => {
+  const [shipping, setShipping] = useState(null)
+  const [loading, setLoading] = useState(false)
+
+  const [getShipping, { data }] = useLazyQuery<
```
Comment (tlgimenes) [inline packages/gatsby-theme-vtex/src/components/ShippingSimulator/index.tsx:72]: should we revalidate this data or do a one shot query ?

File: packages/gatsby-theme-vtex/src/components/ShippingSimulator/ShippingSimulator.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import React, { FC } from 'react'
+import { useIntl } from '@vtex/gatsby-plugin-i18n'
+import { Button, Box } from '@vtex/store-ui'
+import { components } from '@vtex/address-form'
```
[the remark this answers is not in the document]
Comment (lucis) [reply packages/gatsby-theme-vtex/src/components/ShippingSimulator/ShippingSimulator.tsx]: Removed this package!

File: packages/gatsby-theme-vtex/package.json [hunk 4]
```diff
@@ -55,5 +55,6 @@
     "requestidlecallback-polyfill": "^1.0.2",
     "schema-dts": "^0.6.0",
     "swr": "^0.3.0"
-  }
+  },
+  "gitHead": "2d6e385cad076f2072cbe297dd6df03eee6a5470"
```
Comment (lucis) [inline packages/gatsby-theme-vtex/package.json:59]: Remove it before merging

File: packages/gatsby-theme-vtex/src/components/ShippingSimulator/Preview.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React, { FC } from 'react'
+import Skeleton from 'react-loading-skeleton'
+import { Box } from '@vtex/store-ui'
+
+interface Props {
+  variant?: string
+}
+
+const OfferPreview: FC<Props> = ({ variant = '' }) => (
+  <Box variant={`${variant}.preview`} sx={{ marginX: 10 }}>
```
[the remark this answers is not in the document]
Comment (lucis) [reply packages/gatsby-theme-vtex/src/components/ShippingSimulator/Preview.tsx:10]: I'll add it as mandatory, nice catch

**Review by emersonlaurentino — APPROVED** [review]

**Pull request merged**